packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -1018,57 +1039,69 @@ class _ViewContentState extends State<_ViewContent> {
-                                updateSuggestions();
-                              },
-                              onSubmitted: widget.viewOnSubmitted,
-                              textCapitalization: widget.textCapitalization,
-                              textInputAction: widget.textInputAction,
-                              keyboardType: widget.keyboardType,
-                            ),
+              child: OverflowBox(
+                alignment: Alignment.topLeft,
+                maxWidth: math.min(widget.viewMaxWidth, _screenSize!.width),
+                minWidth: 0,
+                fit: OverflowBoxFit.deferToChild,
+                child: FadeTransition(
+                  opacity: viewIconsFadeCurve,
+                  child: Column(
+                    mainAxisSize: MainAxisSize.min,
+                    crossAxisAlignment: CrossAxisAlignment.stretch,
+                    children: <Widget>[
+                      Padding(
+                        padding: EdgeInsets.only(top: widget.topPadding),
+                        child: SafeArea(
+                          top: false,
+                          bottom: false,
+                          child: SearchBar(
+                            autoFocus: true,
+                            constraints: headerConstraints ?? (widget.showFullScreenView ? BoxConstraints(minHeight: _SearchViewDefaultsM3.fullScreenBarHeight) : null),
+                            padding: WidgetStatePropertyAll<EdgeInsetsGeometry?>(effectiveBarPadding),
+                            leading: widget.viewLeading ?? defaultLeading,
+                            trailing: widget.viewTrailing ?? defaultTrailing,
+                            hintText: widget.viewHintText,
+                            backgroundColor: const MaterialStatePropertyAll<Color>(Colors.transparent),
+                            overlayColor: const MaterialStatePropertyAll<Color>(Colors.transparent),
+                            elevation: const MaterialStatePropertyAll<double>(0.0),
+                            textStyle: MaterialStatePropertyAll<TextStyle?>(effectiveTextStyle),
+                            hintStyle: MaterialStatePropertyAll<TextStyle?>(effectiveHintStyle),
+                            controller: _controller,
+                            onChanged: (String value) {
+                              widget.viewOnChanged?.call(value);
+                              updateSuggestions();
+                            },
+                            onSubmitted: widget.viewOnSubmitted,
+                            textCapitalization: widget.textCapitalization,
+                            textInputAction: widget.textInputAction,
+                            keyboardType: widget.keyboardType,
                           ),
                         ),
+                      ),
+                      if (!effectiveShrinkWrap || widget.showFullScreenView || result.isNotEmpty) ...<Widget>[
```

Comment:
The goal of this check is to not show the divider when only the search bar is shown with no results view visible.

- `!effectiveShrinkWrap`: If the view isn't shrink wrapped, then there will always be space under the search bar visible, and thus the divider should be shown as well
- `minHeight > 0`: I just added this check again. If the minHeight is not zero, there will always be space under the search bar, and the divider should be shown
- `widget.showFullScreenView`: In full screen, the view takes up the whole screen, so the divider should be visible
- `result.isNotEmpty`: If there are any results, the results will show and there should be a divider

Are any of these checks redundant? I think I should add back the `viewConstraints.minHeight > 0` check since that is another case where there could be space under the search bar that isn't handled by the existing checks.

Reply:
I think we can just draw it whenever it can. For example, when we set shrinkWrap to true and result is empty, but also set viewConstraints.minHeight > 0, we should see an empty view with a divider, but we don't with this change:)

Reply:
Oops I forgot to commit the change. Now it's there.